prime-router/src/test/kotlin/azure/SettingsFunctionTests.kt

```diff
@@ -0,0 +1,581 @@
+        val httpRequestMessage = setupHttpRequestMessageForTesting()
+        val response = settingsFunction.getOrganizations(httpRequestMessage)
+        assertThat(response.status).isEqualTo(HttpStatus.UNAUTHORIZED)
+    }
+
+    @Test
+    fun `test access DHPrimeAdmins can view all organization settings (getOrganizations)`() {
+        val settingsFunction = setupSettingsFunctionForTesting("DHPrimeAdmins", mockFacade())
+        val httpRequestMessage = setupHttpRequestMessageForTesting()
+        var response = settingsFunction.getOrganizations(httpRequestMessage)
+        assertThat(response.status).isEqualTo(HttpStatus.OK)
+        response = settingsFunction.getOrganizations(httpRequestMessage)
+        assertThat(response.status).isEqualTo(HttpStatus.OK)
+    }
+
+    @Test
+    fun `test access user cannot view their organization's settings (getOneOrganization)`() {
+        val settingsFunction = setupSettingsFunctionForTesting(oktaClaimsOrganizationName, mockFacade())
+        val httpRequestMessage = setupHttpRequestMessageForTesting()
+        val response = settingsFunction.getOneOrganization(
+            httpRequestMessage,
+            "$organizationName.elr-secondary"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.UNAUTHORIZED)
+    }
+
+    @Test
+    fun `test access user cannot view another organization's settings (getOneOrganization)`() {
+        val settingsFunction = setupSettingsFunctionForTesting(oktaClaimsOrganizationName, mockFacade())
+        val httpRequestMessage = setupHttpRequestMessageForTesting()
+        val response = settingsFunction.getOneOrganization(
+            httpRequestMessage,
+            "$otherOrganizationName.test-lab-2"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.UNAUTHORIZED)
+    }
+
+    @Test
+    fun `test access DHPrimeAdmins can view any organization's settings (getOneOrganization)`() {
+        val settingsFunction = setupSettingsFunctionForTesting("DHPrimeAdmins", mockFacade())
+        val httpRequestMessage = setupHttpRequestMessageForTesting()
+        var response = settingsFunction.getOneOrganization(
+            httpRequestMessage,
+            "$organizationName.elr-secondary"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.OK)
+        response = settingsFunction.getOneOrganization(
+            httpRequestMessage,
+            "$otherOrganizationName.test-lab-2"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.OK)
+    }
+
+    @Test
+    fun `test access user cannot update their organization's settings (updateOneOrganization)`() {
+        val settingsFunction = setupSettingsFunctionForTesting(oktaClaimsOrganizationName, mockFacade())
+        val httpRequestMessage = setupHttpRequestMessageForTesting(HttpMethod.PUT)
+        val response = settingsFunction.updateOneOrganization(
+            httpRequestMessage,
+            "$organizationName.elr-secondary"
```

Comment:
Doesn't impact the test, but same issue here, this should be an org not receiver

Reply:
fixed